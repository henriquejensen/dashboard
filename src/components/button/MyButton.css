.my-button-circle {
    padding: 2px 7px 0px 7px;
    background: #F0AD4E;
    color: white;
    border-color: #F0AD4E;
    border-radius: 15px;
}

.my-button-circle:hover {
    color: white;
    background: #f9cd8f;
    border-color: #f9cd8f;
}

.my-button-circle:focus {
    color: white;
    background: #f9cd8f;
    border-color: #f9cd8f;
}