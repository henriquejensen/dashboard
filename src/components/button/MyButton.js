import React from 'react';
import { Button } from "react-bootstrap";

const MyButton = (props) => {
    let tooltip = props.tooltip ? props.tooltip : "";
    let labelButton = props.label ? props.label : "";
    let params = props.params;
    let myButtonClass = props.myButtonClass ? props.myButtonClass : "mapa-button";
    let myButtonStyle = props.myButtonStyle ? props.myButtonStyle : "info";
    let myButtonText = props.myButtonText ? props.myButtonText : <i className='fa fa-search'/>;

    return (
        <span>
            <a data-tip data-for={tooltip}>
                <Button bsStyle={myButtonStyle} className={myButtonClass} onClick={params ? () => props.onClickButton(...params) : props.onClickButton}>
                    {myButtonText}
                </Button>
            </a>
            {labelButton}
        </span>
    );
};

export default MyButton;