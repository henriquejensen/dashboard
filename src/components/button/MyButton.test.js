import React from 'react';
import renderer from "react-test-renderer";
import MyButton from "./MyButton";
import { shallow } from 'enzyme';

describe("<MyButton/>", () => {
    let tooltip="toolTipConsultar";
    let labelButton="Documento"
    
    it('gerando o elemento sem quebrar', () => {
        const tree = renderer.create(<MyButton tooltip={tooltip} label={labelButton} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});