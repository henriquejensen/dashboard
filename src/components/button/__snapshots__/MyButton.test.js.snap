// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyButton/> gerando o elemento sem quebrar 1`] = `
<span>
  <a
    data-for="toolTipConsultar"
    data-tip={true}
  >
    <button
      className="mapa-button btn btn-info"
      disabled={false}
      type="button"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </a>
  Documento
</span>
`;
